SMelidoni/personal-finance-manager
# feature/BE-login-screen

## Description of Changes

Implemented secure user registration and login API. Passwords are hashed before storing in the database and JWT tokens are used for authentication. The JWT secret is stored in .env file for security.

## Type of Change


## Describe the problem this PR solves and your solution.

The problem was the lack of a system for user registration and login. The solution was to create secure APIs for registering and logging in users. 

## Testing Steps

1. Use Postman or a similar tool to send a POST request to the `/auth/register` endpoint. The body of the request should contain `firstName`, `lastName`, `email`, `password` and `confirmPassword` fields. `phone` is optional.
2. Check the database to confirm that the user was created.
3. Send a POST request to the `/auth/login` endpoint with `email` and `password` fields in the body. The response should include a JWT token.
4. Use the JWT token in the `Authorization` header for accessing protected routes.

## Screenshots

N/A

## Checklist
